arangod/Replication2/StateMachines/Document/DocumentCore.h

```diff
@@ -35,19 +35,22 @@ namespace arangodb::replication2::replicated_state::document {
 struct DocumentCore {
   explicit DocumentCore(
       GlobalLogIdentifier gid, DocumentCoreParameters coreParameters,
-      std::shared_ptr<IDocumentStateAgencyHandler> agencyHandler,
-      std::shared_ptr<IDocumentStateShardHandler> shardHandler,
+      std::shared_ptr<IDocumentStateHandlersFactory> const& handlersFactory,
       LoggerContext loggerContext);
 
   LoggerContext const loggerContext;
 
   auto getShardId() -> std::string_view;
 
+  auto getTransactionHandler()
+      -> std::shared_ptr<IDocumentStateTransactionHandler>;
+
  private:
   GlobalLogIdentifier _gid;
   DocumentCoreParameters _params;
   ShardID _shardId;
   std::shared_ptr<IDocumentStateAgencyHandler> _agencyHandler;
   std::shared_ptr<IDocumentStateShardHandler> _shardHandler;
+  std::shared_ptr<IDocumentStateTransactionHandler> _transactionHandler;
```

Comment:
I kinda disagree with this. The transaction handler contains the currently ongoing transactions and this **not** something that should be shared across multiple instances of the document state. For example, if a follower becomes a leader, this structure is still in place. It should be located at in the follower state. Hence it is destroyed and thus all the transactions are gone as well.

The purpose of the core is to hold information that is supposed to be shared across instances. Ongoing transactions are not part of that.

Reply:
Thanks, that's some good insight. I am currently addressing this in an upcoming commit.